### PR TITLE
Updated Codebase and License

### PR DESCRIPTION
**Pull request by sheldonhull**
Hopefully these changes would be good with you. I really enjoyed using this module for a project and thought it could use a few tweaks and noticed a missing license in it. Put it MIT like dbatools and other projects are using and hopefully it would be cool with you. 
- Updated code from the latest changes in psgallery
- Included updated license agreement as no license provided and normally projects created now include them for proper reuse
- error handling put on if menu items null as this resolved issues with invalid index being found. I have option strict on so it might be catching the error that in non-strict mode would be allowable

### CODE REVIEW

File: ps-menu.psm1 [hunk 1]
```diff
@@ -1,87 +1,122 @@
-function DrawMenu {
+write-debug "Loading: ps-menu.psm1"
+
+function DrawMenu
+{
+    [cmdletbinding()]
     param ($menuItems, $menuPosition, $Multiselect, $selection)
+
+	# ensure preferences passed down to continue into function
```
Comment (chrisseroka) [inline ps-menu.psm1:8]: I think this comment does not bring much value, please remove

File: ps-menu.psm1 [hunk 2]
```diff
@@ -1,87 +1,122 @@
...
-					$item = '[ ] ' + $item
-				}
-			}
-			if ($i -eq $menuPosition) {
-				Write-Host "> $($item)" -ForegroundColor Green
-			} else {
-				Write-Host "  $($item)"
-			}
-		}
+    for ($i = 0; $i -le $l; $i++)
```
Comment (chrisseroka) [inline ps-menu.psm1:10]: I see a bug here, there should be: `$i -lt $l` instead (lower than vs lower than or equal)

File: ps-menu.psm1 [hunk 3]
```diff
@@ -1,87 +1,122 @@
...
+    {
+
+        #if ($menuItems[$i] -ne $null)
+        if(@($menuitems).count -gt 0)
```
Comment (chrisseroka) [inline ps-menu.psm1:14]: And then we do not need this line (because of `$i -lt $l`)

File: ps-menu.psm1 [hunk 4]
```diff
@@ -1,87 +1,122 @@
...
-			if ($i -eq $menuPosition) {
-				Write-Host "> $($item)" -ForegroundColor Green
-			} else {
-				Write-Host "  $($item)"
-			}
-		}
+    for ($i = 0; $i -le $l; $i++)
+    {
+
+        #if ($menuItems[$i] -ne $null)
```
Comment (chrisseroka) [inline ps-menu.psm1:13]: leftover

File: ps-menu.psm1 [hunk 5]
```diff
@@ -1,87 +1,122 @@
...
+        if(@($menuitems).count -gt 0)
+        {
+
+            #write-verbose ($menuItems | format-table | out-string) -verbose
```
Comment (chrisseroka) [inline ps-menu.psm1:17]: leftover

File: ps-menu.psm1 [hunk 6]
```diff
@@ -1,87 +1,122 @@
...
+            #write-verbose "`$i: $i" -Verbose
```
Comment (chrisseroka) [inline ps-menu.psm1:18]: leftover (commented code)

File: ps-menu.psm1 [hunk 7]
```diff
@@ -1,87 +1,122 @@
...
+
+            if ($i -le (@($menuItems).count))
```
Comment (chrisseroka) [inline ps-menu.psm1:20]: It's always true because of loop condition (with fix)

**Review by chrisseroka — CHANGES_REQUESTED** [review]

**Comment by sheldonhull** [comment]
Thanks for the response, I'll be more than glad to fix. I just found an invalid array index that showed up after i sent this over. Can I will take your input and just do a new pull request with those fixes as well. cheers!

**Pull request closed without merging**